internal/common/function/function.go

```diff
@@ -77,66 +77,125 @@ func ValidateBackupSpec(nonAdminBackup *nacv1alpha1.NonAdminBackup) error {
 	return nil
 }
 
-// GenerateVeleroBackupName generates a Velero backup name based on the provided namespace and NonAdminBackup name.
-// It calculates a hash of the NonAdminBackup name and combines it with the namespace and a prefix to create the Velero backup name.
-// If the resulting name exceeds the maximum Kubernetes name length, it truncates the namespace to fit within the limit.
-func GenerateVeleroBackupName(namespace, nabName string) string {
-	// Calculate a hash of the name
-	hasher := sha256.New()
-	_, err := hasher.Write([]byte(nabName))
-	if err != nil {
-		return ""
+// GenerateNacObjectNameWithUUID generates a unique name based on the provided namespace and object origin name.
+// It includes a UUID suffix. If the name exceeds the maximum length, it truncates nacName first, then namespace.
+func GenerateNacObjectNameWithUUID(namespace, nacName string) string {
```

Comment:
I would change function implementation to make it more readable (did not use the consts you defined, but they can be used here)
```go
func Generate(namespace, name string) string {
	uuidSuffix := uuid.New().String()

	remainingLength := validation.DNS1123LabelMaxLength - len(uuidSuffix) - len("--")

	if len(namespace+name) > remainingLength {
		if len(namespace) >= remainingLength {
			return fmt.Sprintf("%s-%s", namespace[:remainingLength], uuidSuffix)
		}
		remainingLength := remainingLength - len(namespace)
		return fmt.Sprintf("%s-%s-%s", name[:remainingLength], namespace, uuidSuffix)
	}

	return fmt.Sprintf("%s-%s-%s", namespace, name, uuidSuffix)
}
```

Reply:
I like your implementation better ! Do you want to include in this PR or as follow-up?

Reply:
either way id fine

Reply:
When replacing with your function I got the test failure. Because current function does work in those test scenarios I will live it as is, because I don't want to spent time on fixing your function:

```
--- FAIL: TestGenerateVeleroObjectNameUUID (0.00s)
    --- FAIL: TestGenerateVeleroObjectNameUUID/nabName_empty (0.00s)
        function_test.go:196: Generated name contains double hyphens: example--e949bb26-67ef-479a-a968-47cebc894f28
    --- FAIL: TestGenerateVeleroObjectNameUUID/very_long_name_and_namespace_empty (0.00s)
        function_test.go:196: Generated name contains double hyphens: rrrrrrrrrrrrrrrrrrrrrrrr--094ab4e3-a9af-42b8-ae4c-33f697052753
    --- FAIL: TestGenerateVeleroObjectNameUUID/empty_namespace_and_empty_name (0.00s)
        function_test.go:196: Generated name contains double hyphens: --5a3d285f-2c2c-4707-9081-fc437f2b299e
```